### PR TITLE
new functions added

### PR DESCRIPTION
**Pull request by ZiweiWU713**
I have created flexible girds of 2D and 3D viewing. The user can choose the layout they want and then load the image. In the grids, the image have these function of mouse click:
  left: move
  middle: brightness and contrast
  right: zoom
  mouse wheel: scrolling through slices

If you want to try the GUI, choose Qt_main.py, change "C:/Users/hansw/Videos/artefacts" in line 83 to the folder which contains all the probands folders on you computer(make it easy to find the images), then run Qt_main.py.

I will add more functions later, for example, double click to have a seperate large-size view.

### CODE REVIEW

File: GUI/PyQt/Qt_main.py [hunk 1]
```diff
@@ -243,142 +77,175 @@ def resample(self, image, scan, new_spacing=[1, 1, 1]):
...
-            self.slider.valueChanged.connect(self.View_A)
-            # spinbox
-            #self.valueset.setRange(1, self.slices)
-            #self.valueset.valueChanged.connect(self.slice_update)
-            #self.valueset.setValue(1)
-
-
-    def View_A(self):
-        if self.voxel_ndarray == []:
+            self.PathDicom = QtWidgets.QFileDialog.getExistingDirectory(self, "open file", "C:/Users/hansw/Videos/artefacts")  # root directory
```
Comment (thomaskuestner) [inline GUI/PyQt/Qt_main.py]: The path to the dicom directory can be obtained via 
```
with open('config' + os.sep + 'param.yml', 'r') as ymlfile:
cfg = yaml.safe_load(ymlfile)
dbinfo = DatabaseInfo(cfg['MRdatabase'],cfg['subdirs'])
```

And then `dbinfo.sPathIn`
or 
`os.path.join(dbinfo.sPathIn, pat, dbinfo.sSubDirs[1], seq)`
with `pat = dbinfo.lPats[i]` and `seq = cfg['selectedDatabase']['dataref']` or `seq= cfg['selectedDatabase']['dataart']`
is what you are looking for

File: GUI/PyQt/framework1.py [hunk 2]
```diff
@@ -11,102 +11,163 @@
 class Ui_MainWindow(object):
     def setupUi(self, MainWindow):
         MainWindow.setObjectName("MainWindow")
-        MainWindow.resize(1340, 849)
+        MainWindow.resize(1796, 979)
```
Comment (thomaskuestner) [inline GUI/PyQt/framework1.py]: What if the screen is smaller than the defined size? Can you automatically determine the maximal possible size?
Is there a desirable height/width ratio of the GUI which is looking good or does it work for any ratio?

File: GUI/PyQt/framework1.ui [hunk 3]
```diff
@@ -6,229 +6,329 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>1340</width>
-    <height>849</height>
+    <width>1796</width>
```
Comment (thomaskuestner) [inline GUI/PyQt/framework1.ui]: Is this file fixed or generate each time automatically?
If fixed: Are the values empirical chosen?

File: GUI/PyQt/view3D_coronal.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    new_real_shape = image.shape * resize_factor
+    new_shape = np.round(new_real_shape)
+    real_resize_factor = new_shape / image.shape
+    new_spacing = spacing / real_resize_factor
+
+    image = scipy.ndimage.interpolation.zoom(image, real_resize_factor)
+
+    return image, new_spacing
+
+data_path = "C:\\Users\hansw\Videos\\artefacts\\03\dicom_sorted\\t1_tse_tra_Kopf_Motion_0003"
```
Comment (thomaskuestner) [inline GUI/PyQt/view3D_coronal.py]: Please modify the path. See first comment

File: GUI/PyQt/view3D_coronal.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        self.im3 = self.ax.contourf(np.transpose(self.Z[:, self.ind, :]), hatches=[None, '//', '\\', 'XX'], colors='none', edges='r', levels=np.arange(5), extent=[0,196,132,0])
+
+        #plt.xlim(0, voxel_ndarray.shape[0] * dx)
+        #plt.ylim(voxel_ndarray.shape[1] * dy, 0)
+        plt.axis('off')
+        self.ax.set_ylabel('slice %s' % self.ind)
+        self.fig.canvas.draw_idle()
+
+#fig, ax = plt.subplots(1, 1)    ### 2 output?
+PathDicom = "C:\\Users\hansw\Videos\\artefacts\\03\dicom_sorted\\t1_tse_tra_Kopf_Motion_0003"
```
Comment (thomaskuestner) [inline GUI/PyQt/view3D_coronal.py]: Modify the path

File: GUI/PyQt/Qt_main.py (hunk 1 above)
Comment (ZiweiWU713) [reply GUI/PyQt/Qt_main.py]: I add this into my code, now it works. But this setting only fit for the path in our labor. If the user want to use the GUI on other computer, they must make change in the 27th line of DatabaseInfo.py :

self.sPathIn = '/med_data/ImageSimilarity/Databases'+ os.sep + self.sDatabase + os.sep + self.sSubDirs[0]

 '/med_data/ImageSimilarity/Databases' needs to be changed to the path which contains the database.

File: GUI/PyQt/view3D_coronal.py (hunk 4 above)
Comment (ZiweiWU713) [reply GUI/PyQt/view3D_coronal.py]: This file and the next one are only functions of my 3D multiclass color coding. I will integrate it later into my GUI after I finish all the overlay functions.

**Pull request merged**